File: ios/ObjectiveC/ObjectiveC/Beacon/4-Alert/Complete/beaconAlertComplete/ViewController.m
//
//  ViewController.m
//  beaconAlertStart
//
//  Created by sarra srairi on 29/03/2016.
//  Copyright © 2016 sarra srairi. All rights reserved.
//

#import "ViewController.h"
#import "AlertViewControllerAction.h"
#import "FirstWayAlertTimerStrategy.h"
@interface ViewController () {
 
    ATBeaconContent *currentAlertBeaconContent;
}

@end

@implementation ViewController

-(void)viewDidLoad {
    [super viewDidLoad];
 
    // Do any additional setup after loading the view, typically from a nib.
}

-(void)viewDidAppear:(BOOL)animated{
    [super viewDidAppear:animated];
    //register the protocol for did range beacon
    [[ATBeaconManager sharedInstance] registerBeaconAlertDelgate:self];
 
}
-(void)viewDidDisappear:(BOOL)animated{
    [super viewDidDisappear:animated];
    [[ATBeaconManager sharedInstance] registerBeaconAlertDelgate:nil];
}

-(void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

//comment Alert delegate method
-(BOOL)createBeaconAlert:(ATBeaconContent *)_beaconContent{
   
    if([@"popup" isEqualToString:[_beaconContent getAction]]){
        // create a popup view
        NSLog(@"Well done! now you can create your Pop UP");
        _txtAlertMessage.text = [_beaconContent getAlertTitle];
        [_txtAlertMessage setNeedsDisplay];
        _buttonAlert.hidden=false;
        currentAlertBeaconContent = _beaconContent;
        return YES ;
    }
    
    _txtAlertMessage.text = @"No Beacons ready for action";
    [_txtAlertMessage setNeedsDisplay];
    return NO ;
}
 
-(BOOL)removeBeaconAlert:(ATBeaconContent *)_beaconContent actionStatus:(ATBeaconRemoveStatus)_actionStatus{
    _buttonAlert.hidden= true;
    _txtAlertMessage.text = @"Remove beacon alert action";
    [_txtAlertMessage setNeedsDisplay];
    return YES;
}

-(void)onNetworkError:(ATRangeFeedStatus)_feedStatus{
}


-(void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender{
    if ([segue.identifier isEqualToString:@"alert"]) {
        AlertViewControllerAction *controllerAlert= (AlertViewControllerAction *)[segue destinationViewController];
        controllerAlert.alertBeaconContent = currentAlertBeaconContent;
    }
}

@end

File: ios/SWIFT/Beacon/9-Alert-Strategy/Complete/AlertStrategy/ViewController.swift
//
//  ViewController.swift
//  AlertStrategy
//
//  Created by sarra srairi on 11/08/2016.
//  Copyright © 2016 R&D connecthings. All rights reserved.
//

import UIKit
import ATAnalytics
import ATLocationBeacon

class ViewController: UIViewController, ATBeaconAlertDelegate {
    
    @IBOutlet weak var txtAlertMessage: UILabel!
    
    @IBOutlet weak var actionTxt: UILabel!
    var currentAlertBeaconContent: ATBeaconContent!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        ATBeaconManager.sharedInstance().registerBeaconAlertDelgate(self)
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    func createBeaconAlert(_ _beaconContent: ATBeaconContent!) -> Bool {
        if ("popup" == _beaconContent.getAction()) {
            // create a popup view
            actionTxt.text = "Well done! now you have created your alert action"
            self.txtAlertMessage.text = _beaconContent.getAlertTitle()
            txtAlertMessage.setNeedsDisplay()
            
            currentAlertBeaconContent = _beaconContent
            return true
        }
        self.txtAlertMessage.text = "No Beacons ready for action"
        txtAlertMessage.setNeedsDisplay()
        return false
        
    }
    
    func removeBeaconAlert(_ _beaconContent: ATBeaconContent!, actionStatus _actionStatus: ATBeaconRemoveStatus) -> Bool {
        actionTxt.text = "Well done! now you have removed your alert action"
        self.txtAlertMessage.text = "Remove beacon alert action"
        txtAlertMessage.setNeedsDisplay()
        return true
        
    }
    
    func onNetworkError(_ _feedStatus: ATRangeFeedStatus) {
        
    }
    
}

